impl/openapi/src/main/java/io/serverlessworkflow/impl/executors/openapi/OpenAPIExecutor.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright 2020-Present The Serverless Workflow Specification Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.serverlessworkflow.impl.executors.openapi;
+
+import io.serverlessworkflow.api.types.CallHTTP;
+import io.serverlessworkflow.api.types.CallOpenAPI;
+import io.serverlessworkflow.api.types.Endpoint;
+import io.serverlessworkflow.api.types.EndpointUri;
+import io.serverlessworkflow.api.types.TaskBase;
+import io.serverlessworkflow.api.types.UriTemplate;
+import io.serverlessworkflow.api.types.Workflow;
+import io.serverlessworkflow.impl.TaskContext;
+import io.serverlessworkflow.impl.WorkflowApplication;
+import io.serverlessworkflow.impl.WorkflowContext;
+import io.serverlessworkflow.impl.WorkflowException;
+import io.serverlessworkflow.impl.WorkflowModel;
+import io.serverlessworkflow.impl.executors.CallableTask;
+import io.serverlessworkflow.impl.executors.http.HttpExecutor;
+import io.serverlessworkflow.impl.expressions.ExpressionDescriptor;
+import io.serverlessworkflow.impl.expressions.ExpressionFactory;
+import io.serverlessworkflow.impl.resources.ResourceLoader;
+import jakarta.ws.rs.core.UriBuilder;
+import java.net.URI;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
+
+public class OpenAPIExecutor implements CallableTask<CallOpenAPI> {
+
+  private CallOpenAPI task;
+  private Workflow workflow;
+  private WorkflowApplication application;
+  private TargetSupplier targetSupplier;
+
+  private ResourceLoader resourceLoader;
+
+  @Override
+  public boolean accept(Class<? extends TaskBase> clazz) {
+    return clazz.equals(CallOpenAPI.class);
+  }
+
+  @Override
+  public CompletableFuture<WorkflowModel> apply(
+      WorkflowContext workflowContext, TaskContext taskContext, WorkflowModel input) {
+    String operationId = task.getWith().getOperationId();
+    URI openAPIEndpoint = targetSupplier.apply(workflowContext, taskContext, input);
+    OpenAPIProcessor processor = new OpenAPIProcessor(operationId, openAPIEndpoint);
+    OperationDefinition operation = processor.parse();
+
+    OperationPathResolver pathResolver =
+        new OperationPathResolver(
+            operation.getPath(),
+            application,
+            task.getWith().getParameters().getAdditionalProperties());
```

Comment:
I think all this stuff should be moved to init method (thy are always the same for different executions, so they only need to be processed once)